File: src/main/scala/resonant/engine/wrapper/RenderTileDummy.scala
package resonant.engine.wrapper

import resonant.engine.spatial.block.SpatialBlock
import resonant.engine.component.render.RenderDynamic
import net.minecraft.tileentity.TileEntity
import universalelectricity.core.transform.vector.Vector3

/**
 * @author Calclavia
 */
@SideOnly(Side.CLIENT)
class RenderTileDummy extends TileEntitySpecialRenderer
{
  def renderTileEntityAt(tile: TileEntity, x: Double, y: Double, z: Double, f: Float)
  {
    if (tile.isInstanceOf[SpatialBlock])
    {
      val spatial = tile.asInstanceOf[SpatialBlock]
      spatial.components.get(classOf[RenderDynamic]).foreach(_.renderDynamic(new Vector3(x, y, z), false, f))
    }
  }
}

File: src/main/scala/resonant/engine/wrapper/RenderTileDummy.java
package resonant.engine.wrapper;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;
import universalelectricity.core.transform.vector.Vector3;

/**
 * @author Calclavia
 */
@SideOnly(Side.CLIENT)
public class RenderTileDummy extends TileEntitySpecialRenderer
{
	@Override
	public void renderTileEntityAt(TileEntity tile, double x, double y, double z, float f)
	{
		if (tile instanceof TileBlock)
		{
			TileBlock tileBlock = ((TileBlock) tile);

			if (tileBlock.getRenderer() != null)
			{
				tileBlock.getRenderer().renderDynamic(new Vector3(x, y, z), false, f);
			}
		}
	}
}
